cgdb: fix escape sequences

meator opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64-glibc)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

The current version of cgdb has broken escape sequences (its prompt is `[?2004h(gdb) `). This is already fixed in their Github repo, but the latest official release does not contain the fix. The issue mentioning this is [here](https://github.com/cgdb/cgdb/issues/256). I have created the `fixescape.patch` patch from [these](https://github.com/cgdb/cgdb/issues/239#issuecomment-706249589) patches and from [this](https://github.com/cgdb/cgdb/issues/255#issuecomment-820546156) patch.

ericonr commented:
Please include the same information about patch origins into the commit message ;)

meator commented:
I have added the info in the commit description. I have also changed the patch to make the [transition to -Np1](https://github.com/void-linux/void-packages/pull/31550) easier.
